core/src/main/java/com/datastax/oss/driver/internal/core/addresstranslation/SubnetAddressTranslator.java

```diff
@@ -0,0 +1,226 @@
+ * that proxy is different for nodes located in different subnets (eg. when Cassandra is deployed in
+ * multiple datacenters/regions). One can use this, for example, for Cassandra on Kubernetes with
+ * different Cassandra datacenters deployed to different Kubernetes clusters.
+ */
+public class SubnetAddressTranslator implements AddressTranslator {
+
+  private static final Logger LOG = LoggerFactory.getLogger(SubnetAddressTranslator.class);
+
+  /**
+   * A map of Cassandra node subnets (CIDR notations) to target addresses, for example (note quoted
+   * keys):
+   *
+   * <pre>
+   * advanced.address-translator.subnet-addresses {
+   *   "100.64.0.0/15" = "cassandra.datacenter1.com:9042"
+   *   "100.66.0.0/15" = "cassandra.datacenter2.com"
+   *   # IPv6 example:
+   *   # "::ffff:6440:0/111" = "cassandra.datacenter1.com:9042"
+   *   # "::ffff:6442:0/111" = "cassandra.datacenter2.com"
+   * }
+   * </pre>
+   *
+   * If configured without port, the default 9042 will be used. Also supports IPv6 subnets. Note:
+   * subnets must be represented as prefix blocks, see {@link inet.ipaddr.Address#isPrefixBlock()}.
+   */
+  public static final String ADDRESS_TRANSLATOR_SUBNET_ADDRESSES =
+      "advanced.address-translator.subnet-addresses";
+
+  /**
+   * A default address to fallback to if Cassandra node IP isn't contained in any of the configured
+   * subnets. If configured without port, the default 9042 will be used. Also supports IPv6
+   * addresses.
+   */
+  public static final String ADDRESS_TRANSLATOR_DEFAULT_ADDRESS =
+      "advanced.address-translator.default-address";
+
+  public static DriverOption ADDRESS_TRANSLATOR_SUBNET_ADDRESSES_OPTION =
+      new DriverOption() {
+        @NonNull
+        @Override
+        public String getPath() {
+          return ADDRESS_TRANSLATOR_SUBNET_ADDRESSES;
+        }
+      };
+
+  public static DriverOption ADDRESS_TRANSLATOR_DEFAULT_ADDRESS_OPTION =
+      new DriverOption() {
+        @NonNull
+        @Override
+        public String getPath() {
+          return ADDRESS_TRANSLATOR_DEFAULT_ADDRESS;
+        }
+      };
+
+  private static final String DELIMITER = ":";
+  private static final int DEFAULT_PORT = 9042;
+
+  private final List<SubnetAddress> subnetAddresses;
+  private final Optional<InetSocketAddress> defaultAddress;
+  private final String logPrefix;
+
+  public SubnetAddressTranslator(@NonNull DriverContext context) {
+    logPrefix = context.getSessionName();
+    this.subnetAddresses =
+        context.getConfig().getDefaultProfile()
+            .getStringMap(ADDRESS_TRANSLATOR_SUBNET_ADDRESSES_OPTION).entrySet().stream()
+            .map(
+                e -> {
+                  // Quoted and/or containing forward slashes map keys in reference.conf are read to
+                  // strings with additional quotes, eg. 100.64.0.0/15 -> '100.64.0."0/15"' or
+                  // "100.64.0.0/15" -> '"100.64.0.0/15"'
```

Comment:
This is a bit of a weird idiosyncrasy of the parser, but like you said, it's invisible to the user, so I think it's ok.